ndcube/ndcube.py

```diff
@@ -230,6 +230,43 @@ def array_axis_physical_types(self):
         return [tuple(world_axis_physical_types[axis_correlation_matrix[:, i]])
                 for i in range(axis_correlation_matrix.shape[1])][::-1]
 
+    def _generate_pixel_grid(self, edges):
+        # Create meshgrid of all pixel coordinates.
+        # If user, wants edges, set pixel values to pixel edges.
+        # Else make pixel centers.
+        wcs_shape = self.data.shape[::-1]
+        if edges:
+            wcs_shape = tuple(np.array(wcs_shape) + 1)
+            ranges = [np.arange(i) - 0.5 for i in wcs_shape]
+        else:
+            ranges = [np.arange(i) for i in wcs_shape]
+
+        return np.meshgrid(*ranges, indexing='ij', sparse=True)
+
+    def _calculate_world_indices(self, axes):
```

Comment:
This should probably be named something more like `_calculate_world_axes` or `_calculate_world_axis_indices`.  When I started reading the code I was expecting it to get the indices to input to `world_to_array_index` based on the name.  But instead it seems to be getting the indices of the axes in world order.

Reply:
Thinking about it, perhaps this should be a util function rather than a method?  All you need is `axes` and `axis_correlation_matrix`.  `world_n_dim` can be derived from `axis_correlation_matrix`.